src/Microsoft.Health.Fhir.Shared.Core/Features/Search/BundleFactory.cs

```diff
@@ -63,8 +64,27 @@ public ResourceElement CreateHistoryBundle(SearchResult result)
                     Method = hasVerb ? httpVerb : null,
                     Url = hasVerb ? $"{r.Resource.ResourceTypeName}/{(httpVerb == Bundle.HTTPVerb.POST ? null : r.Resource.ResourceId)}" : null,
                 };
+
+                string statusString;
+                switch (httpVerb)
+                {
+                    case Bundle.HTTPVerb.POST:
+                        statusString = ((int)HttpStatusCode.Created).ToString() + " " + HttpStatusCode.Created;
+                        break;
+                    case Bundle.HTTPVerb.PUT:
+                        statusString = ((int)HttpStatusCode.OK).ToString() + " " + HttpStatusCode.OK;
+                        break;
+                    case Bundle.HTTPVerb.DELETE:
+                        statusString = ((int)HttpStatusCode.NoContent).ToString() + " " + HttpStatusCode.NoContent;
+                        break;
+                    default:
+                        statusString = ((int)HttpStatusCode.OK).ToString() + " " + HttpStatusCode.OK;
```

Comment:
>statusString = ((int)HttpStatusCode.OK).ToString() + " " + HttpStatusCode.OK; [](http://example.com/codeflow?start=24&length=77)

I would throw exception here.

If we get something we don't expect, it's dangerous to hide it, and better to fail execution.